internal/pkg/instrumentation/probe/probe.go

```diff
@@ -280,6 +258,58 @@ func (i *Base[BPFObj, BPFEvent]) Close() error {
 	return err
 }
 
+type SpanProducer[BPFObj any, BPFEvent any] struct {
+	Base[BPFObj, BPFEvent]
+
+	Version   string
```

Comment:
Currently, this version is constant across all the probes that use `SpanProducer` - Do we still want to pass it through all the `New` functions?

Reply:
I think we do. The end goal is to have custom probes, each of which will need to report its own version. If we design our probes to match this end goal we will avoid having to remove the coupling added when liking the version information here.

Reply:
I'm going to resolve this. We can re-address in a follow-up PR if the need arises.